variables.tf

```
@@ -1,9 +1,605 @@
+    setting = string
+  })
+
+  default = null
+
+  validation {
+    condition     = var.additional_unattend_content == null ? true : local.is_windows
+    error_message = "Additional unanttend content can only be defined for Windows virtual machines."
+  }
+}
+
+variable "admin_password" {
+  description = "Password to use for the local administrator on this virtual machine. If not set, a password will be generated and stored in the Key Vault specified by key_vault_id."
+  default     = null
+  type        = string
+}
+
+variable "admin_ssh_public_key" {
+  description = "Public key to use for SSH authentication. Must be at least 2048-bit and in ssh-rsa format."
+  default     = null
+  type        = string
+}
+
+variable "admin_username" {
+  default     = "azureadmin"
+  description = "Username of the local administrator for the virtual machine."
+  type        = string
+}
+
+variable "allow_extension_operations" {
+  description = "Should Extension Operations be allowed on this Virtual Machine?"
+  type        = bool
+  default     = true
+}
+
+variable "authentication_type" {
+  description = "Specifies the authentication type to use. Valid options are `SSH` or `Password`. Windows virtual machines support only `Password`."
+  default     = "Password"
   type        = string
 
   validation {
-    condition     = length(var.example_variable) >= 3 && length(var.example_variable) <= 13
-    error_message = "Example variable must be between 3 and 13 characters"
+    condition     = var.authentication_type == "Password" || (var.authentication_type == "SSH" && !local.is_windows)
+    error_message = "On Windows operating systems, authentication_type = \"SSH\" is not supported. Use authentication_type = \"Password\" for Windows images."
   }
 }
+
+variable "availability_set_id" {
+  description = "Specifies the ID of the Availability Set in which the Virtual Machine should exist"
+
+  type    = string
+  default = null
+}
+
+variable "backup_policy_id" {
+  description = "The ID of the backup policy to use."
+  type        = string
+  default     = null
+
+  validation {
+    condition     = (var.enable_backup_protected_vm && var.backup_policy_id != null) || !var.enable_backup_protected_vm
+    error_message = "A backup policy ID is required when backup_protected_vm.enabled is true."
+  }
+}
+
+variable "boot_diagnostics" {
+  description = <<-EOT
+    Enable boot diagnostics and optionally specify the storage account to use to store boot diagnostics. The default is to use a managed storage account to store boot diagnostics when enabled.
+
+    Optional parameters:
+
+    Parameter | Description
+    -- | --
+    `enable` | Whether to enable (`true`) or disable (`false`) boot diagnostics.
+    `storage_account_uri` | The endpoint for the Azure storage account that should be used to store boot diagnostics, including console output and hypervisor screenshots.
+  EOT
+
+  type = object({
+    enable              = optional(bool, true)
```

For consistency reasons, it should be named enabled.
```suggestion
    enabled              = optional(bool, true)
```